rotkehlchen/exchanges/exchange.py

```diff
@@ -230,14 +230,31 @@ def query_online_income_loss_expense(
         Has to be implemented by exchanges if they have anything exchange specific
 
         For example coinbase
+
+        Deprecated, will be replaced by query_online_history_events
         """
-        raise NotImplementedError(
-            'query_online_income_loss_expense should only be implemented by subclasses',
-        )
+        return []
 
-    def query_history_events(self) -> None:
+    def query_online_history_events(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> Sequence['HistoryBaseEntry']:
+        """Queries the exchange's API for history events of the user
+
+        Should be implemented in subclasses.
+        Returns events based on HistoryBaseEntry, such as HistoryEvent, AssetMovement, etc.
+        """
+        # Should raise NotImplementedError here after all exchanges are updated
```

Comment:
What's this?

Reply:
This is no longer applicable. As mentioned here: https://github.com/rotki/rotki/pull/9042#discussion_r1876303157 `query_history_events` may be reimplemeted with custom logic that won't necessarily use this function, which means this function may not need implemented in all subclasses after all.

But to explain what it was supposed to mean - this comment was simply to remind us that once all exchanges have been updated and implement this function, it should do this here:
```
raise NotImplementedError(
    'query_online_history_events() should only be implemented by subclasses',
)
```
Like is done in `query_online_margin_history` for example, to prevent the addition of exchanges without implementing this function.